src/event_loop.rs

```diff
@@ -64,21 +62,24 @@ pub struct EventLoop {
 
 impl EventLoop {
     pub fn spawn() -> Self {
-        let (tx, rx) = mpsc::unbounded_channel::<EventLoopMsg>();
+        let (tx, mut rx) = mpsc::unbounded_channel::<EventLoopMsg>();
 
         let j = unwrap!(thread::Builder::new()
             .name("QuicP2p-Event-Loop".into())
             .spawn(move || {
-                let event_loop_future = rx.map_err(|_| ()).for_each(move |ev_loop_msg| {
-                    if let Some(mut f) = ev_loop_msg.0 {
-                        f();
-                        Ok(())
-                    } else {
-                        Err(())
+                let event_loop_future = async move {
+                    while let Some(ev_loop_msg) = rx.recv().await {
+                        if let Some(mut f) = ev_loop_msg.0 {
+                            f();
+                        } else {
+                            break;
+                        }
                     }
-                });
+                };
+
+                let mut runtime = unwrap!(tokio::runtime::Runtime::new());
```

Comment:
Here if we cannot start runtime we may need to panic, but with a clear message. We can add description to unwrap here AFAIK `unwrap!(tokio::runtime::Runtime::new(), "Cannot start quick-p2p runtime, aborting")